src/VisualStudio/Core/Def/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -91,6 +92,7 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         private readonly IVsService<IBrokeredServiceContainer> _brokeredServiceContainer;
         private readonly AsynchronousOperationListenerProvider _listenerProvider;
         private readonly RemoteServiceCallbackDispatcherRegistry _callbackDispatchers;
+        private readonly TaskCompletionSource<bool> _clientCreationSource = new();
```

Comment:
```suggestion
        private readonly TaskCompletionSource<bool> _clientCreationSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
```

Otherwise continuations might run inline here and not sure if that'd be too surprising?